internal/core/http.go

```diff
@@ -26,10 +27,24 @@ type HTTPRedirectFilter struct {
 	Status   int
 }
 
+type URLRewriteType int
+
+const (
+	ReplaceFullPathURLRewriteType URLRewriteType = iota
+	ReplacePrefixMatchURLRewriteType
+)
+
+type HTTPURLRewriteFilter struct {
+	Type               URLRewriteType
+	ReplaceFullPath    string
+	ReplacePrefixMatch string
+}
```

Comment:
(I know I added this 😆 ), but I'd remove full rewrite since we won't support it in the scope of this PR

```suggestion
const (
	ReplacePrefixMatchURLRewriteType = iota
)

type HTTPURLRewriteFilter struct {
	Type               URLRewriteType
	ReplacePrefixMatch string
}
```